runtime/executor/src/main/java/edu/snu/onyx/runtime/executor/data/blocktransfer/BlockInputStream.java

```diff
@@ -112,7 +110,11 @@ void append(final ByteBuf byteBuf) {
    */
   void markAsEnded() {
     byteBufInputStream.byteBufQueue.close();
-    completeFuture.complete(inputStreamIterator);
+    try {
+      completeFuture.complete(new DataUtil.InputStreamIterator<>(byteBufInputStream, serializer));
```

Comment:
Let's construct `InputStreamIterator` in `setSerializer` like what we did before.
It is dangerous to throw `RuntimeException` in netty thread.